File: sunbin/day16/001_숨어있는숫자의덧셈.js
// https://school.programmers.co.kr/learn/courses/30/lessons/120851

function solution(my_string) {
  const numbers = my_string
                  .replace(/[^0-9]/g, '')
                  .split('')
                  .reduce((acc, curr) => {
                    return acc + Number(curr)
                  }, 0);

  return numbers

}

solution("aAb1B2cC34oOp")

File: sunbin/Lv0/day16/python/001_숨어있는숫자의덧셈.py
# https://school.programmers.co.kr/learn/courses/30/lessons/120851

import re


def solution(my_string):
    numbers = re.sub('[^0-9]', '', my_string)
    answer = sum([int(i) for i in numbers])
    
    return answer


# 다른 풀이: isdigit()
def solution_b(my_string):
    return sum(int(i) for i in my_string if i.isdigit())